examples/variational_autoencoder.py

```diff
@@ -5,33 +5,30 @@
 import numpy as np
 import matplotlib.pyplot as plt
 
-from keras.layers import Input, Dense, Lambda
+from keras.layers import Input, Dense, merge
 from keras.models import Model
 from keras import backend as K
 from keras import objectives
 from keras.datasets import mnist
 
-batch_size = 16
+batch_size = 100
 original_dim = 784
 latent_dim = 2
-intermediate_dim = 128
-epsilon_std = 0.01
-nb_epoch = 40
+intermediate_dim = 500
+nb_epoch = 100
 
 x = Input(batch_shape=(batch_size, original_dim))
 h = Dense(intermediate_dim, activation='relu')(x)
 z_mean = Dense(latent_dim)(h)
-z_log_std = Dense(latent_dim)(h)
+z_log_var = Dense(latent_dim)(h)
 
 def sampling(args):
-    z_mean, z_log_std = args
-    epsilon = K.random_normal(shape=(batch_size, latent_dim),
-                              mean=0., std=epsilon_std)
-    return z_mean + K.exp(z_log_std) * epsilon
+    z_mean, z_log_var = args
+    epsilon = K.random_normal(shape=(batch_size, latent_dim), mean=0.)
+    return z_mean + K.exp(z_log_var/2) * epsilon
 
 # note that "output_shape" isn't necessary with the TensorFlow backend
-# so you could write `Lambda(sampling)([z_mean, z_log_std])`
-z = Lambda(sampling, output_shape=(latent_dim,))([z_mean, z_log_std])
+z = merge([z_mean, z_log_var], mode=sampling, output_shape=(latent_dim,))
```

Comment:
I like the previous version too! Unfortunately, it doesn't run in keras 1.0.5 because this version contains the line `self.supports_masking = True` for Lambda. Should we just leave it be for when keras 1.0.6 rolls out?